src/expr/src/scalar/func.rs

```diff
@@ -2825,6 +2825,42 @@ fn text_concat_variadic<'a>(datums: &[Datum<'a>], temp_storage: &'a RowArena) ->
     Datum::String(temp_storage.push_string(buf))
 }
 
+fn pad_leading<'a>(
+    datums: &[Datum<'a>],
+    temp_storage: &'a RowArena,
+) -> Result<Datum<'a>, EvalError> {
+    let string = datums[0].unwrap_str();
+
+    let len = match usize::try_from(datums[1].unwrap_int64()) {
+        Ok(len) => len,
+        Err(_) => {
+            return Err(EvalError::InvalidParameterValue(
+                "length must be positive".to_owned(),
+            ))
+        }
+    };
+
+    let pad_string = if datums.len() == 3 {
+        datums[2].unwrap_str()
+    } else {
+        " "
+    };
+
+    let mut buf: String;
+    if len < string.len() {
+        buf = string[0..len].to_string();
+    } else {
+        buf = pad_string
+            .chars()
```

Comment:
Similar issue to the above. In the `else` clause, you are correctly iterating by character, but in the `if` clause, you are taking the first `len` bytes, so you would get an invalid answer if your string contains any multi-byte Unicode characters. See https://users.rust-lang.org/t/how-to-get-a-substring-of-a-string/1351/3